File: Adapter/AdapterPattern.py
from abc import ABCMeta, abstractmethod

class Target:
    __metaclass__ = ABCMeta
    
    @abstractmethod
    def target_method(self):
        pass

class Adaptee:
    def spec_method(self):
        print("spec_method in Adaptee")

class Adapter(Target):
    def __init__(self, adaptee):
        self._adaptee = adaptee

    def target_method(self):
        self._adaptee.spec_method()

if __name__ == "__main__":
    adaptee = Adaptee()
    adapter = Adapter(adaptee)
    adapter.target_method()

File: Adapter/AdapterPattern.java
interface Target {
    public void targetMethod();
}

class Adaptee {
    public void specMethod() {
        System.out.println("specMethod in Adaptee");
    }
}

class Adapter implements Target {
    private Adaptee adaptee;

    public Adapter(Adaptee adaptee) {
        this.adaptee = adaptee;
    }

    public void targetMethod() {
        adaptee.specMethod();
    }
}

public class AdapterPattern {
    public static void main(String[] args) {
        Adaptee adaptee = new Adaptee();
        Adapter adapter = new Adapter(adaptee);
        adapter.targetMethod();
    }
}

